File: share/pegasus/examples/simple-examples/split.java
import edu.isi.pegasus.planner.dax.*;

public class split {

    public static void main(String[] args) {
        ADAG dax = new ADAG("split");

        File webpage = new File("pegasus.html");

        Job curl = new Job("ID1", "curl");
        curl.addArgument("-o").addArgument(webpage);
        curl.addArgument("http://pegasus.isi.edu");
        curl.uses(webpage, File.LINK.OUTPUT, File.TRANSFER.FALSE, false);
        dax.addJob(curl);

        Job split = new Job("ID2", "split");
        split.addArgument("-l 100");
        split.addArgument("-a 1");
        split.addArgument(webpage);
        split.addArgument("part.");
        split.uses(webpage, File.LINK.INPUT, File.TRANSFER.FALSE, false);
        dax.addJob(split);

        dax.addDependency(curl, split);

        String[] array = new String[] { "a","b","c","d" };
        for (String c : array) {
            File part = new File("part." + c);
            split.uses(part, File.LINK.OUTPUT, File.TRANSFER.FALSE, false);

            File count = new File("count.txt." + c);

            Job wc = new Job("ID"+c, "wc");
            wc.addArgument("-l");
            wc.addArgument(part);
            wc.setStdout(count);
            wc.uses(part, File.LINK.INPUT);
            wc.uses(count, File.LINK.OUTPUT, File.TRANSFER.TRUE, false);
            dax.addJob(wc);

            dax.addDependency(split, wc);
        }

        dax.writeToSTDOUT();
    }
}

File: share/pegasus/examples/simple-examples/split.py
#!/usr/bin/env python3

import sys
from Pegasus.DAX3 import *

dax = ADAG("split")

webpage = File("pegasus.html")

curl = Job("curl")
curl.addArguments("-o",webpage,"http://pegasus.isi.edu")
curl.uses(webpage, link=Link.OUTPUT)
dax.addJob(curl)

split = Job("split")
split.addArguments("-l","100","-a","1",webpage,"part.")
split.uses(webpage, link=Link.INPUT)
dax.addJob(split)

dax.depends(split, curl)

for c in "abcd":
    part = File("part.%s" % c)
    split.uses(part, link=Link.OUTPUT)

    count = File("count.txt.%s" % c)

    wc = Job("wc")
    wc.addArguments("-l",part)
    wc.setStdout(count)
    wc.uses(part, link=Link.INPUT)
    wc.uses(count, link=Link.OUTPUT, transfer=True, register=False)
    dax.addJob(wc)

    dax.depends(wc, split)

dax.writeXML(sys.stdout)

